demo-app/build.gradle.kts

```diff
@@ -2,17 +2,20 @@ import java.io.FileInputStream
 import java.util.Properties
 
 plugins {
-    id("otel.android-app-conventions")
+    id("com.android.application") version "8.4.0"
+    id("org.jetbrains.kotlin.android") version "1.9.23"
 }
 
 val localProperties = Properties()
 localProperties.load(FileInputStream(rootProject.file("local.properties")))
 
 android {
     namespace = "io.opentelemetry.android.demo"
+    compileSdk = 34
 
     defaultConfig {
```

Comment:
In notice the default targetSdk for the extension project is 23 - I'll ask more about that in Slack, but for the app, I think we should target 34 as well. Not that we'll ever submit this to the Play Store, but it requires the targetSdk to be at least your compileSdk if your target is less than 34, so the current combination won't be legal under Play Store rules.

Reply:
I also commented on your other PR @bidetofevil but now that I see this, it makes more sense. I think with demo-app being its own stand-alone project (this PR) we can target 34. Seems reasonable.